doc/xml/parsing.md

```diff
@@ -0,0 +1,504 @@
+]>
+DOCTYPE
+doc = Nokogiri::XML.parse(xml)
+doc
+# =>
+#(Document: {
+  name = "document",
+  children = [
+    #(DTD: {
+      name = "note",
+      children = [
+        #(ElementDecl: { "<!ELEMENT payment (#PCDATA)>\n" }),
+        #(AttributeDecl: { "<!ATTLIST payment type CDATA \"check\">\n" })]
+      })]
+  })
+```
+
+### Conditional Sections
+
+\Nokogiri parses an [XML conditional section](https://www.w3.org/TR/REC-xml/#sec-condition-sect)
+into a Nokogiri::XML::EntityDecl object:
+
+```
+xml = <<DOCTYPE
+<!DOCTYPE note [
+<!ENTITY % draft 'INCLUDE' >
+]>
+DOCTYPE
+doc = Nokogiri::XML.parse(xml)
+doc
+# =>
+#(Document: {
+  name = "document",
+  children = [
+    #(DTD: {
+      name = "note",
+      children = [ #(EntityDecl: { "<!ENTITY % draft \"INCLUDE\">\n" })]
+      })]
+  })
+```
+
+### Character References
+
+\Nokogiri parses an [XML character reference](https://www.w3.org/TR/REC-xml/#sec-references)
+(such as <tt>&#9792;</tt>)
+and replaces it with a character such as (<tt>'♀'</tt>):
+
+```
+xml = <<ELE
+<root>
+  <name>
+    <vorname>Marie</vorname>
+    <nachname>M&#252;ller</nachname>
+    <geschlecht>&#9792;</geschlecht>
+  </name>
+</root>
+ELE
+doc = Nokogiri::XML.parse(xml)
+doc
+# =>
+#(Document: {
+  name = "document",
+  children = [
+    #(Element: {
+      name = "root",
+      children = [
+        #(Text "\n  "),
+        #(Element: {
+          name = "name",
+          children = [
+            #(Text "\n    "),
+            #(Element: { name = "vorname", children = [ #(Text "Marie")] }),
+            #(Text "\n    "),
+            #(Element: { name = "nachname", children = [ #(Text "Müller")] }),
+            #(Text "\n    "),
+            #(Element: { name = "geschlecht", children = [ #(Text "♀")] }),
+            #(Text "\n  ")]
+          }),
+        #(Text "\n")]
+      })]
+  })
+```
+
+### Entity References
+
+\Nokogiri parses an [XML entity reference](https://www.w3.org/TR/REC-xml/#sec-references)
+(such as <tt>&lt;</tt>)
+and replaces it with text such as (<tt>'<'</tt>):
+
+```
+xml = '<root>An entity reference is needed for the less-than character (&lt;).</root>'
+doc = Nokogiri::XML.parse(xml)
+doc
+# =>
+#(Document: {
+  name = "document",
+  children = [
+    #(Element: {
+      name = "root",
+      children = [ #(Text "An entity reference is needed for the less-than character (<).")]
+      })]
+  })
+```
+
+### Entity Declarations
+
+\Nokogiri parses an [XML entity declaration](https://www.w3.org/TR/REC-xml/#sec-entity-decl)
+into a Nokogiri::XML::EntityDecl object:
+
+```
+xml = <<DTD
+<!DOCTYPE note [
+  <!ENTITY company "Example Corp">
+]>
+DTD
+doc = Nokogiri::XML.parse(xml)
+doc
+# =>
+#(Document: {
+  name = "document",
+  children = [
+    #(DTD: {
+      name = "note",
+      children = [ #(EntityDecl: { "<!ENTITY company \"Example Corp\">\n" })]
+      })]
+  })
+```
+
+### Text Declaration
+
+\Nokogiri parses an [XML text declaration](https://www.w3.org/TR/REC-xml/#sec-TextDecl)
```

Comment:
I don't think this needs a separate section, it's a subset of what you document in the XML declaration section.

Reply:
Agree that it's redundant, but I included it b/c that's the way it is in the W3C doc.  Your call (of course).

Reply:
Removed.